presto-base-arrow-flight/src/test/java/com/facebook/plugin/arrow/TestArrowFlightEchoQueries.java

```diff
@@ -0,0 +1,555 @@
+                mapWriter.endMap();
+            }
+            mapWriter.setValueCount(numValues);
+
+            try (VectorSchemaRoot expectedRoot = new VectorSchemaRoot(Arrays.asList(intVector, mapVector));
+                    FlightClient client = FlightClient.builder(bufferAllocator, getServerLocation()).verifyServer(false).build()) {
+                expectedRoot.setRowCount(numValues);
+
+                String tableName = "map";
+                addTableToServer(client, expectedRoot, tableName);
+
+                assertQueryFails(format("SELECT * FROM %s", tableName), "MapType is currently unsupported");
+
+                removeTableFromServer(client, tableName);
+            }
+        }
+    }
+
+    @Test
+    public void testStructVector() throws Exception
+    {
+        try (BufferAllocator bufferAllocator = allocator.newChildAllocator("echo-test-client", 0, Long.MAX_VALUE);
+                IntVector intVector = new IntVector("id", bufferAllocator);
+                StructVector structVector = StructVector.empty("struct", bufferAllocator)) {
+            MaterializedResult.Builder expectedBuilder = resultBuilder(getSession(), INTEGER,
+                    RowType.from(ImmutableList.of(
+                            new RowType.Field(Optional.of("int"), INTEGER),
+                            new RowType.Field(Optional.of("long"), BIGINT))));
+
+            final IntVector childIntVector
+                    = structVector.addOrGet("int", FieldType.nullable(new ArrowType.Int(32, true)), IntVector.class);
+            final BigIntVector childLongVector
+                    = structVector.addOrGet("long", FieldType.nullable(new ArrowType.Int(64, true)), BigIntVector.class);
+            childIntVector.allocateNew();
+            childLongVector.allocateNew();
+
+            final int numValues = 10;
+            for (int i = 0; i < numValues; i++) {
+                intVector.setSafe(i, i);
+                childIntVector.setSafe(i, i + i);
+                childLongVector.setSafe(i, i * i);
+                structVector.setIndexDefined(i);
+                expectedBuilder.row(i, ImmutableList.of(i + i, (long) i * i));
+            }
+
+            try (VectorSchemaRoot expectedRoot = new VectorSchemaRoot(Arrays.asList(intVector, structVector));
+                    FlightClient client = FlightClient.builder(bufferAllocator, getServerLocation()).verifyServer(false).build()) {
+                expectedRoot.setRowCount(numValues);
+
+                String tableName = "structs";
+                addTableToServer(client, expectedRoot, tableName);
+
+                MaterializedResult actual = computeActual(format("SELECT * FROM %s", tableName));
+
+                assertEquals(actual.getRowCount(), numValues);
+                assertEquals(actual, expectedBuilder.build());
+
+                removeTableFromServer(client, tableName);
+            }
+        }
+    }
+
+    @Test
+    public void testDictionaryVector() throws Exception
```

Comment:
I moved dictionary block testing to this test, instead of defining a separate Flight server just for a single test